azure-samples/src/main/java/com/microsoft/azure/management/network/samples/ManageNetworkWatcher.java

```diff
@@ -0,0 +1,337 @@
+            System.out.println("Creating virtual machine...");
+            VirtualMachine vm = azure.virtualMachines().define(vmName)
+                    .withRegion(region)
+                    .withExistingResourceGroup(rgName)
+                    .withNewPrimaryNetwork(virtualNetworkDefinition)
+                    .withPrimaryPrivateIPAddressDynamic()
+                    .withNewPrimaryPublicIPAddress(dnsLabel)
+                    .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_14_04_LTS)
+                    .withRootUsername(userName)
+                    .withRootPassword("Abcdef.123456")
+                    .withSize(VirtualMachineSizeTypes.STANDARD_A1)
+                    .defineNewExtension("packetCapture")
+                        .withPublisher("Microsoft.Azure.NetworkWatcher")
+                        .withType("NetworkWatcherAgentLinux")
+                        .withVersion("1.4")
+                        .withMinorVersionAutoUpgrade()
+                        .attach()
+                    .create();
+
+            // Create storage account
+            System.out.println("Creating storage account...");
+            StorageAccount storageAccount = azure.storageAccounts().define(saName)
+                    .withRegion(region)
+                    .withExistingResourceGroup(rgName)
+                    .create();
+
+            // Start a packet capture
+            System.out.println("Creating packet capture...");
+            PacketCapture packetCapture = nw.packetCaptures()
+                    .define(packetCaptureName)
+                    .withTarget(vm.id())
+                    .withStorageAccountId(storageAccount.id())
+                    .withTimeLimitInSeconds(1500)
+                    .definePacketCaptureFilter()
+                        .withProtocol(PcProtocol.TCP)
+                        .attach()
+                    .create();
+            System.out.println("Created packet capture");
+            Utils.print(packetCapture);
+
+            // Stop a packet capture
+            System.out.println("Stopping packet capture...");
+            packetCapture.stop();
+            Utils.print(packetCapture);
+
+            // Get a packet capture
+            System.out.printf("Getting packet capture...");
+            PacketCapture packetCapture1 = nw.packetCaptures().getByName(packetCaptureName);
+            Utils.print(packetCapture1);
+
+            // Delete a packet capture
+            System.out.println("Deleting packet capture");
+            nw.packetCaptures().deleteByName(packetCapture.name());
+
+            //============================================================
+            // Verify IP flow – verify if traffic is allowed to or from a virtual machine
+            // Get the IP address of a NIC on a virtual machine
+            String ipAddress = vm.getPrimaryNetworkInterface().ipConfigurations().values().iterator().next().privateIPAddress();
```

Comment:
instead of looking for the first IP config on a NIC, `NetworkInterface.primaryIPConfiguration()` should be used here. I'll deal with this after merging this PR.